http-server/src/main/java/xtdb/jackson/TxDeserializer.java

```diff
@@ -0,0 +1,56 @@
+package xtdb.jackson;
+
+import clojure.lang.*;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.*;
+import com.fasterxml.jackson.databind.deser.std.StdDeserializer;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import xtdb.IllegalArgumentException;
+import xtdb.tx.Ops;
+import xtdb.tx.Tx;
+
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.util.ArrayList;
+import java.util.List;
+
+public class TxDeserializer extends StdDeserializer<Tx>  {
+
+    public TxDeserializer() {
+        super(Tx.class);
+    }
+
+    @Override
+    public Tx deserialize(com.fasterxml.jackson.core.JsonParser jp, DeserializationContext ctxt) throws IOException, JsonProcessingException {
+        ObjectMapper mapper = (ObjectMapper) jp.getCodec();
+        ObjectNode node = mapper.readTree(jp);
+        List<Ops> ops = new ArrayList<>();
+        LocalDateTime systemTime = null;
+        ZoneId defaultTz = null;
+
+        try {
+            if (node.has("tx_ops")) {
+                // The reason that we are fully walking the nodes is that Jsonista overwrites the standard `List` parser and
```

Comment:
are we getting any benefit from the Jsonista Clojure module now, or is it worth us going straight to Jackson? i.e. are we ever deserialising any Clojure data structures in this API?

Reply:
Yes, when we deserializing literals.